BuildTasks/Common/Common.ts

```diff
@@ -82,11 +82,19 @@ export function setTfxManifestArguments(tfx: ToolRunner): (() => void) {
         if (isPreview) { jsonOverrides.galleryFlags = ["Preview"]; };
     }
 
-    const extensionVersion = tl.getInput("extensionVersion", false);
+    let extensionVersion = tl.getInput("extensionVersion", false);
     if (extensionVersion) {
-        tl.debug(`Overriding extension version to: ${extensionVersion}`);
-        jsonOverrides = (jsonOverrides || {});
-        jsonOverrides.version = extensionVersion;
+       const extractedVersions = extensionVersion.match(/[0-9]+\.[0-9]+\.[0-9]+/);
+       if (extractedVersions && extractedVersions.length === 1) {
+            extensionVersion = extractedVersions[0];
+            tl.debug(`Overriding extension version to: ${extensionVersion}`);
+            jsonOverrides = (jsonOverrides || {});
+            jsonOverrides.version = extensionVersion;
+        }
+        else
```

Comment:
I think we need to tweak the regex in order to handle `extensionVersion` values such as `1.2.1.3` including a 4th part. There are some extensions using that format, and we need to preserve them and not transform them in 1.2.1.  

As soon as we have this extract version logic, shouldn't we also apply the same logic while determining the `Major`, `Minor` and `Patch` version used in build tasks when the user chooses to "Override task version"?. Shouldn't the version extracted here to be propagated to the build tasks when "Override tasks version"?